src/main/java/duke/storage/Storage.java

```diff
@@ -0,0 +1,140 @@
+package duke.storage;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.Todo;
+
+/**
+ * The Storage class implements a repository that deals with the
+ * loading of tasks from the file and saving tasks in the file.
+ *
+ * @author Chia Jeremy
+ */
+public class Storage {
+
+    private final File file;
+
+    /**
+     * Class constructor for storage.
+     *
+     * @param filePath the location of the storage file
+     */
+    public Storage(String filePath) {
+        this.file = new File(filePath);
+    }
+
+    /**
+     * Returns the saved tasks as a list.
+     *
+     * @return the saved tasks as a list
+     */
+    public List<String> load() {
+        return getSavedTasksAsList();
+    }
+
+    /**
+     * Adds the task into storage.
+     *
+     * @param task the task to be added into storage
+     */
+    public void add(Task task) {
+        try {
+            FileWriter fw = new FileWriter(this.file, true);
+            String data;
+            if (task instanceof Todo) {
+                Todo td = (Todo) task;
+                data = "T |   | " + td.getDescription();
+            } else if (task instanceof Deadline) {
+                Deadline dl = (Deadline) task;
+                data = "D |   | " + dl.getDescription() + " | " + dl.getDateTime();
+            } else {
+                Event evt = (Event) task;
+                data = "E |   | " + evt.getDescription() + " | " + evt.getStartDt() + " | " + evt.getEndDt();
+            }
+            fw.write(data + System.lineSeparator());
+            fw.flush();
+            fw.close();
+        } catch (IOException e) {
+            System.out.println("\"Fail to add to repository.");
+        }
+    }
+
+    /**
+     * Mark the saved task.
+     *
+     * @param index the index of the task to mark
+     */
+    public void mark(int index) {
+        List<String> tasks = getSavedTasksAsList();
+        StringBuilder sb = new StringBuilder(tasks.get(index));
+        sb.setCharAt(4, 'X');
```

Comment:
Is this a magic number? I think it could have been assigned with varible name such as startingIndex.